alchemy/src/cloudflare/snippet.ts

```diff
@@ -0,0 +1,466 @@
+   * The identifying name of the snippet
+   * Must be unique within the zone
+   * Only lowercase letters (a-z), numbers (0-9), and underscores (_) are allowed
+   */
+  name: string;
+
+  /**
+   * The JavaScript code content of the snippet
+   */
+  content: string;
+
+  /**
+   * Whether to delete the snippet
+   * If set to false, the snippet will remain but the resource will be removed from state
+   * @default true
+   */
+  delete?: boolean;
+
+  /**
+   * Whether to adopt an existing snippet
+   * @default false
+   */
+  adopt?: boolean;
+
+  /**
+   * The Cloudflare-generated zone ID (only used for update/delete operations)
+   * @internal
+   */
+  zoneId?: string;
+
+  /**
+   * The snippet name (only used for update/delete operations)
+   * @internal
+   */
+  snippetId?: string;
+}
+
+/**
+ * Snippet rule for controlling when a snippet executes
+ */
+export interface SnippetRule {
+  /**
+   * The expression that defines when this rule applies
+   * Uses Cloudflare's Rules language
+   */
+  expression: string;
+
+  /**
+   * The name of the snippet to execute when this rule matches
+   */
+  snippetName: string;
+
+  /**
+   * Description of what this rule does
+   */
+  description?: string;
+
+  /**
+   * Whether this rule is enabled
+   * @default true
+   */
+  enabled?: boolean;
+}
+
+/**
+ * Output returned after Snippet creation/update.
+ * IMPORTANT: The type name MUST match the exported resource name.
+ */
+export type Snippet = Omit<SnippetProps, "zone" | "content" | "adopt"> & {
+  /**
+   * The ID of the resource
+   */
+  id: string;
+
+  /**
+   * The identifying name of the snippet
+   */
+  name: string;
+
+  /**
+   * The snippet name (used for update/delete operations)
+   * @internal
+   */
+  snippetId: string;
+
+  /**
+   * The zone ID this snippet belongs to
+   */
+  zoneId: string;
+
+  /**
+   * When the snippet was created
+   */
+  createdOn: Date;
+
+  /**
+   * When the snippet was last modified
+   */
+  modifiedOn: Date;
+
+  /**
+   * Resource type identifier
+   * @internal
+   */
+  type: "snippet";
+};
+
+/**
+ * Creates and manages Cloudflare Snippets for executing custom JavaScript at the edge.
+ *
+ * Snippets allow you to run custom JavaScript code on Cloudflare's edge network,
+ * enabling modifications to requests and responses for your zone.
+ *
+ * @example
+ * // Create a basic snippet that adds a custom header
+ * const headerSnippet = await Snippet("custom-header", {
+ *   zone: myZone,
+ *   name: "add-custom-header",
+ *   content: `
+ *     export default {
+ *       async fetch(request) {
+ *         const response = await fetch(request);
+ *         response.headers.set('X-Custom-Header', 'Hello from Snippet');
+ *         return response;
+ *       }
+ *     }
+ *   `
+ * });
+ *
+ * @example
+ * // Create a snippet that modifies request URLs
+ * const urlRewriteSnippet = await Snippet("url-rewrite", {
+ *   zone: "example.com",
+ *   name: "rewrite-api-urls",
+ *   content: `
```

Comment:
in Worker, it's `script | entrypoint`. I kinda regret those names but maybe keep it consistent